File: src/main/dotnet/Controllers/WeatherForecastController.cs
using System.Threading;
using System.Threading.Tasks;
using MediatR;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using se.clouds.app.javanet.domain.weatherforecast;
using se.clouds.app.javanet.domain.weatherforecast.queries;

namespace se.clouds.app.javanet.Controllers
{
    [ApiController]
    [Route("[controller]")]
    public class WeatherForecastController : ControllerBase
    {
        private readonly IMediator _mediator;
        private readonly ILogger<WeatherForecastController> _logger;
        public WeatherForecastController(IMediator mediator, ILogger<WeatherForecastController> logger)
        {
            _mediator = mediator;
            _logger = logger;
        }

        [HttpGet]
        public async Task<WeatherForecast[]> GetAsync()
        {
            return await _mediator.Send(new GetWeatherForecast(), new CancellationToken());
        }
    }
}

File: Components/app/src/main/java/se/clouds/javanet/app/controllers/WeatherForecastController.java
package se.clouds.javanet.app.controllers;

import se.clouds.javanet.app.domain.weatherforecast.query.GetWeatherForecast;
import se.clouds.javanet.core.controller.AppController;
import se.clouds.javanet.core.controller.IActionResult;
import se.clouds.javanet.core.di.Di;
import se.clouds.javanet.core.mediator.MediatR;

public class WeatherForecastController extends AppController
{
    private MediatR<IActionResult> mediatr = (MediatR)Di.GetMediator();
    private static String routePath = "/weatherforecast";

    public WeatherForecastController()
    {
        super(routePath);
    }

    public IActionResult Get()
    {
        var result = mediatr.SendRequest(new GetWeatherForecast());
        return result.orElseThrow();
    }
}
